### PR TITLE
Improve device auto-detection

### PR DESCRIPTION
**Pull request by zpcore**
Resolves https://github.com/pytorch/xla/issues/7730.

Now device are automatically detected when we import torch_xla:
```
$ python
Python 3.10.13 | packaged by conda-forge | (main, Dec 23 2023, 15:36:39) [GCC 12.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch_xla
WARNING:root:libtpu.so and TPU device found. Setting PJRT_DEVICE=TPU.
>>> torch_xla.real_devices()
['TPU:0', 'TPU:1', 'TPU:2', 'TPU:3']
>>> torch_xla.device()
device(type='xla', index=0)
```

### CODE REVIEW

**Comment by zpcore** [comment]
RFC: Is there a plan to hard code PJRT_DEVICE in the python script to replace the PJRT_DEVICE environment variable? @miladm

**Review by will-cromar — COMMENTED** [review]
Thanks for the clean up!

@JackCaoG do you see any problem selecting `$PJRT_DEVICE` when we `import torch_xla`?

File: torch_xla/runtime.py [hunk 1]
```diff
@@ -53,14 +53,11 @@ def set_device_type(pjrt_device: str) -> None:
   os.environ[xenv.PJRT_DEVICE] = pjrt_device
 
 
-def _maybe_select_default_device():
+def maybe_select_default_device():
```
[the remark this answers is not in the document]
Comment (zpcore) [reply torch_xla/runtime.py:56]: I removed it because this is called within a different file (torch_xla.__init__.py). I can fix it back.

**Comment by JackCaoG** [comment]
> Thanks for the clean up!
> 
> @JackCaoG do you see any problem selecting `$PJRT_DEVICE` when we `import torch_xla`?

seems fine to me as long as we don't init the runtime yet.

File: torch_xla/runtime.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (will-cromar) [reply torch_xla/runtime.py]: Fair enough. Feel free to move this to `__init__.py` or `_internal` if you'd like. I just don't want to make this seem like a public function.

**Review by will-cromar — APPROVED** [review]

**Pull request merged**